### PR TITLE
sql: Added incompatibilities with MySQL

### PR DESCRIPTION
**Pull request by morgo**
events, sys do not exist
pfs exists but returns empty.

### CODE REVIEW

**Comment by queenypingcap** [comment]
@shenli  PTAL.

**Comment by morgo** [comment]
I've added more content for beginners (explaining that ENGINE=myisam is not creating a MyISAM table).

**Comment by morgo** [comment]
Added large transactions and EXPLAIN (differs substantially).
Added Optimizer Trace (TiDB has query tracing, which is different from optimizer tracing).

File: sql/mysql-compatibility.md [hunk 1]
```diff
@@ -83,12 +90,20 @@ TiDB implements the asynchronous schema changes algorithm in F1. The Data Manipu
...
 TiDB implements an optimistic transaction model. Unlike MySQL, which uses row-level locking to avoid write conflict, in TiDB, the write conflict is checked only in the `commit` process during the execution of the statements like `Update`, `Insert`, `Delete`, and so on.
 
 **Note:** On the business side, remember to check the returned results of `commit` because even there is no error in the execution, there might be errors in the `commit` process.
 
+### Large transactions
+
+Due to the distributed, 2-phase commit requirement of TiDB, large transactions that modify data can be particularly problematic. TiDB intentionally sets some limits on transaction sizes to reduce this impact:
+
+* Each Key-Value entry is no more than 6MB
+* The total number of Key-Value entries is no more than 300,000 rows
```
Comment (shenli) [inline sql/mysql-compatibility.md]: I suggest removing the `rows`. Because this limitation only considers kv pairs, not sql rows.

**Pull request merged**